packages/browser/src/attributes/UserIdentifierAttributeProvider.ts

```diff
@@ -0,0 +1,28 @@
+import { BacktraceAttributeProvider, BacktraceConfiguration, IdGenerator } from '@backtrace/sdk-core';
+
+export class UserIdentifierAttributeProvider implements BacktraceAttributeProvider {
+    private readonly USER_IDENTIFIER = 'backtrace-guid';
+    private _guid: string | undefined;
+
+    constructor(options: BacktraceConfiguration) {
+        this._guid = options.userAttributes?.['guid'] as string;
+    }
+
+    public get type(): 'scoped' | 'dynamic' {
+        return 'scoped';
+    }
+    public get(): Record<string, unknown> {
+        if (!this._guid) {
+            let guid = window.localStorage.getItem(this.USER_IDENTIFIER);
+            if (!guid) {
+                guid = IdGenerator.uuid();
+                window.localStorage.setItem(this.USER_IDENTIFIER, guid);
```

Comment:
We may want to consider setting localStorage in a try catch as an exception can be thrown if the user configures the browser to not persist data. [https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage#:~:text=Thrown%20in%20one,from%20persisting%20data.](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage#:~:text=Thrown%20in%20one,from%20persisting%20data.)

Reply:
Added thanks!